app/ante/eip712.go

```diff
@@ -217,9 +218,10 @@ func VerifySignature(
 			return errorsmod.Wrap(errortypes.ErrUnknownExtensionOptions, "tx doesnt contain expected amount of extension options")
 		}
 
-		extOpt, ok := opts[0].GetCachedValue().(*ethermint.ExtensionOptionsWeb3Tx)
-		if !ok {
-			return errorsmod.Wrap(errortypes.ErrUnknownExtensionOptions, "unknown extension option")
+		var extOpt *ethermint.ExtensionOptionsWeb3Tx
+		err = registry.UnpackAny(opts[0], &extOpt)
```

Comment:
left a comment here regarding this: https://github.com/dymensionxyz/dymension/issues/761#issuecomment-2030153423